addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2257,7 +2273,20 @@ var SnippetsMenu = Widget.extend({
             this._mutex.exec(() => this._destroyEditor(snippetEditor));
         }
         this._mutex.exec(() => {
-            if (this._currentTab === this.tabs.OPTIONS && !this.snippetEditors.length) {
+            const selection = this.$body[0].ownerDocument.getSelection();
+            const range = selection?.rangeCount && selection.getRangeAt(0);
+            const currentlySelectedNode = range?.commonAncestorContainer;
+            // In some cases (e.g. in translation mode) it's possible to have
+            // all snippet editors destroyed after disabling text options.
+            // We still want to keep the toolbar available in this case.
+            const isEditabTextElementSelected =
+                currentlySelectedNode?.nodeType === Node.TEXT_NODE &&
+                !!currentlySelectedNode?.parentNode?.isContentEditable;
+            if (
+                this._currentTab === this.tabs.OPTIONS &&
+                !this.snippetEditors.length &&
+                !isEditabTextElementSelected
+            ) {
```

Comment:
- Weird `isEditab` variable
- Nitpicking but don't do:
```javascript
A = X()
if (B && A) {

}
```
Do
```javascript
if (B && X()) {

}
```
If you want to make the `X` part more understandable, then the comment can be inside the `if`

Or you can do 
```javascript
if (B) {
    A = X();
    if (A)  {
    
    }
}
```